internal/service/labelstore/service.go

```diff
@@ -122,8 +127,16 @@ func (s *service) Data() any {
 	return s
 }
 
+func (s *service) Enable() {
+	s.enabled.Store(true)
+}
+
 // GetOrAddLink is called by a remote_write endpoint component to add mapping and get back the global id.
 func (s *service) GetOrAddLink(componentID string, localRefID uint64, lbls labels.Labels) uint64 {
+	if !s.enabled.Load() {
```

Comment:
Technically its a slightly different behavior. If you started without a remote write, and just a prometheus write queue. Then changed the config to remove the write queue and add a remote write the ref id's would be a bit odd. That being said I doubt that will ever happen. If it does then it is unlikely to be a problem since the hash is likely to be a much higher number than the incrementing ref id will get and the scraper only uses it to cache data quicker. Additionally since remote_write and prometheus.write.queue cant coexist so we are never in a mixed environment.

Reply:
I could probably add globalref id, but then would need to place it in the lock or make it an atomic which doesnt feel great given how incredibly unlikely the scenario is.

Reply:
On thinking about having the value be consistent is more important so Hash is the best approach.